### PR TITLE
[16.0][ADD] sale_order_general_discount_fixed: add module

### PR DESCRIPTION
**Pull request by tendil**
Task: 5379

## Summary by CodeRabbit

* **New Features**
  * Add a "General Fixed Discount" monetary field on sale orders that proportionally distributes a fixed order-level discount across order lines and recalculates when lines change.

* **Tests**
  * Add tests covering single/multi-line distribution, rounding behavior, zero-priced lines, line deletions, and validation against excessive discounts.

* **Documentation**
  * Add README, usage guide, static HTML docs, and contributor/credits notes.

### CODE REVIEW

File: sale_order_general_discount_fixed/tests/test_sale_order_general_discount_fixed.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+    def test_discount_cannot_exceed_order_amount(self):
+        with self.assertRaises(ValidationError):
+            form = Form(self.env["sale.order"])
+            form.partner_id = self.partner
+            with form.order_line.new() as line:
+                line.product_id = self.product_100
+                line.product_uom_qty = 1
+            form.general_discount_fixed = 150.0
+            form.save()
```
[the remark this answers is not in the document]
Comment (tendil) [reply sale_order_general_discount_fixed/tests/test_sale_order_general_discount_fixed.py:287]: not necessary for now; the test scope for coverage was specified in the task requirements

**Review by Aldeigja — APPROVED** [review]
Functional LGTM
